instrumentation/spring/spring-batch-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/spring/batch/v3_0/chunk/TracingChunkExecutionListener.java

```diff
@@ -28,6 +29,7 @@ public TracingChunkExecutionListener(Class<?> builderClass) {
   }
 
   @Override
+  @Initializer
```

Comment:
I forget what this does?

Reply:
is that true in this case? I'm guessing this listener is created, and then beforeChunk is called multiple times?

Reply:
`chunkContextAndBuilder` is needed by `afterChunk`, and `beforeChunk` is setting it, it's initializing the variable

```java
	/**
	 * Callback before the chunk is executed, but inside the transaction.
	 *
	 * @param context The current {@link ChunkContext}
	 */
	void beforeChunk(ChunkContext context);

	/**
	 * Callback after the chunk is executed, outside the transaction.
	 *
	 * @param context The current {@link ChunkContext}
	 */
	void afterChunk(ChunkContext context);
```

Reply:
that doesn't sound like the expected use of `@Initializer`

Reply:
> Initializer methods are assumed by NullAway
> * to always be called before any other method of the class that is not a constructor or called from
> * a constructor

right, there are other methods like `getOrder` that can be called before - I'll change